7x TestWebKitAPI.iOSMouseSupport* (api-tests) are constant failures on some builds

hortont424 opened the pull request:
#### a524d52158ab97104f3f92cf32f400541ce3a54c
<pre>
7x TestWebKitAPI.iOSMouseSupport* (api-tests) are constant failures on some builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=289599">https://bugs.webkit.org/show_bug.cgi?id=289599</a>
<a href="https://rdar.apple.com/145178841">rdar://145178841</a>

Reviewed by Abrar Rahman Protyasha and Wenson Hsieh.

Some UIKit changes have resulted in a UIApp being required in order to manually
drive gesture recognizers, like we do in WKMouseTouchGestureRecognizer.
Factor out UIApp initialization and adopt it in all of these tests.

Additionally, these changes result in Cancelled state getting folded into Failed,
so adjust one assertion to accept either state for now.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ClipboardTests.mm:
(createWebViewForClipboardTests):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PasteboardUtilities.mm:
(createWebViewWithCustomPasteboardDataEnabled):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAttachmentTests.mm:
(TestWebKitAPI::TEST(WKAttachmentTestsIOS, InsertPastedMapItemAsAttachment)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
* Tools/TestWebKitAPI/Tests/ios/ActionSheetTests.mm:
(TestWebKitAPI::TEST(ActionSheetTests, DISABLED_DismissingActionSheetShouldNotDismissPresentingViewController)):
(TestWebKitAPI::TEST(ActionSheetTests, CopyImageElementWithHREFAndTitle)):
(TestWebKitAPI::TEST(ActionSheetTests, CopyImageElementWithHREF)):
(TestWebKitAPI::TEST(ActionSheetTests, CopyImageElementWithoutHREF)):
(TestWebKitAPI::TEST(ActionSheetTests, CopyLinkWritesURLAndPlainText)):
* Tools/TestWebKitAPI/Tests/ios/UIPasteboardTests.mm:
(TestWebKitAPI::setUpWebViewForPasteboardTests):
* Tools/TestWebKitAPI/cocoa/TestCocoa.h:
* Tools/TestWebKitAPI/cocoa/TestCocoa.mm:
(TestWebKitAPI::Util::instantiateUIApplicationIfNeeded):
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(setOverriddenApplicationKeyWindow):
* Tools/TestWebKitAPI/ios/IOSMouseEventTestHarness.mm:
(TestWebKitAPI::MouseEventTestHarness::mouseCancel):
* Tools/TestWebKitAPI/ios/UserInterfaceSwizzler.h:

Canonical link: <a href="https://commits.webkit.org/292056@main">https://commits.webkit.org/292056@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3b267e8e2c115ee9ea8b5b9bf617cb55ebfc95de

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94776 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14365 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4203 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99792 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45265 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/96823 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14642 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22784 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72297 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29600 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97777 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10923 "Found 1 new test failure: fast/forms/ios/focus-input-via-button-no-scaling.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85573 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52629 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10610 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3285 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44605 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/80844 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3386 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101835 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21804 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15931 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81296 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22051 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81597 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80675 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25235 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2642 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15037 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15222 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21781 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21441 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/24912 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23180 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
LGTM, I wonder if iOS mouse support tests should all derive from a common test fixture class that performs `instantiateUIApplicationIfNeeded()` during `testing::Test::SetUp()`?

hortont424 committed:
7x TestWebKitAPI.iOSMouseSupport* (api-tests) are constant failures on some builds
https://bugs.webkit.org/show_bug.cgi?id=289599
rdar://145178841

Reviewed by Abrar Rahman Protyasha and Wenson Hsieh.

Some UIKit changes have resulted in a UIApp being required in order to manually
drive gesture recognizers, like we do in WKMouseTouchGestureRecognizer.
Factor out UIApp initialization and adopt it in all of these tests.

Additionally, these changes result in Cancelled state getting folded into Failed,
so adjust one assertion to accept either state for now.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ClipboardTests.mm:
(createWebViewForClipboardTests):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PasteboardUtilities.mm:
(createWebViewWithCustomPasteboardDataEnabled):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAttachmentTests.mm:
(TestWebKitAPI::TEST(WKAttachmentTestsIOS, InsertPastedMapItemAsAttachment)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
* Tools/TestWebKitAPI/Tests/ios/ActionSheetTests.mm:
(TestWebKitAPI::TEST(ActionSheetTests, DISABLED_DismissingActionSheetShouldNotDismissPresentingViewController)):
(TestWebKitAPI::TEST(ActionSheetTests, CopyImageElementWithHREFAndTitle)):
(TestWebKitAPI::TEST(ActionSheetTests, CopyImageElementWithHREF)):
(TestWebKitAPI::TEST(ActionSheetTests, CopyImageElementWithoutHREF)):
(TestWebKitAPI::TEST(ActionSheetTests, CopyLinkWritesURLAndPlainText)):
* Tools/TestWebKitAPI/Tests/ios/UIPasteboardTests.mm:
(TestWebKitAPI::setUpWebViewForPasteboardTests):
* Tools/TestWebKitAPI/cocoa/TestCocoa.h:
* Tools/TestWebKitAPI/cocoa/TestCocoa.mm:
(TestWebKitAPI::Util::instantiateUIApplicationIfNeeded):
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(setOverriddenApplicationKeyWindow):
* Tools/TestWebKitAPI/ios/IOSMouseEventTestHarness.mm:
(TestWebKitAPI::MouseEventTestHarness::mouseCancel):
* Tools/TestWebKitAPI/ios/UserInterfaceSwizzler.h:

Canonical link: https://commits.webkit.org/292056@main
